opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -16,9 +18,10 @@ use super::{
 
 /// The storage for sums.
 struct ValueMap<T: Number<T>> {
-    values: Mutex<HashMap<AttributeSet, T>>,
+    buckets: Arc<[Mutex<Option<HashMap<AttributeSet, T>>>; 256]>,
```

Comment:
using hashing function from [ahash](https://crates.io/crates/ahash) crates help get more juice. I think we can have optional dependency on this crate - so use DefaultHasher from std collection by default,  and ahash::AHasher if application includes this crates.